iowa_alcohol_sales.sql

```diff
@@ -19,21 +19,34 @@ from
   order by 1 asc, sum(bottles_sold) desc)
 where Ranking=1;
 
---Cửa hàng nênbán nhiều mặt hàng hay tập trung vào bán ít mặt hàng thì bán được nhiều hơn
+--Cửa hàng nên bán nhiều mặt hàng hay tập trung vào bán ít mặt hàng thì bán được nhiều hơn
+with aggregate_by_store as ( -- Note: Nên viết dưới dạng CTE để dễ xem, dễ truy ngược và chỉnh sửa query nếu phải sửa lỗi
+  select
+    store_number,
+    upper(store_name) as store_name,
+    count(distinct item_number) as number_of_items,
+    round(sum(bottles_sold),0) as number_of_bottles_sold,
+    -- Note: Bán được nhiều hơn có thể hiểu là bán được số lượng sản phẩm (number_of_bottles_sold) nhiều hơn hoặc bán được nhiều đơn (number_of_orders) hơn.
+    count(1) as number_of_orders
+  from `bigquery-public-data.iowa_liquor_sales.sales`
+  group by 1,2
+)
+
 select
-  store_number,
-  upper(store_name) as Store,
-  count(distinct item_number) as No_items,
-  round(sum(sale_dollars),0) as Sales,
-  round(sum(sale_dollars)/count(distinct item_number),0) as SalesPerItems
-FROM `bigquery-public-data.iowa_liquor_sales.sales`
-group by 1,2
-order by 5 desc;
+  *
+  , rank() over (order by number_of_items desc) as rank_number_of_items
```

Comment:
Mấy cái rank này k cần partition by gì à?

Reply:
không cần bởi vì ở đây mình muốn rank các store theo metric, mà ở bảng này store là to nhất rồi